SoundSwitch/Framework/Profile/ProfileManager.cs

```diff
@@ -0,0 +1,194 @@
+            _profileByApplication =
+                AppConfigs.Configuration.ProfileSettings
+                    .Where((setting) => setting.ApplicationPath != null)
+                    .ToDictionary(setting => setting.ApplicationPath.ToLower());
+            _profileByHotkey = AppConfigs.Configuration.ProfileSettings
+                .Where((setting) => setting.HotKeys != null)
+                .ToDictionary(setting => setting.HotKeys);
+        }
+
+        /// <summary>
+        /// Initialize the profile manager. Return the list of Profile that it couldn't register hotkeys for.
+        /// </summary>
+        /// <returns></returns>
+        public Result<ProfileSetting[], VoidSuccess> Init()
+        {
+            RegisterEvents();
+
+            var errors = AppConfigs.Configuration.ProfileSettings
+                .Where(setting => setting.HotKeys != null)
+                .Where(profileSetting => !WindowsAPIAdapter.RegisterHotKey(profileSetting.HotKeys))
+                .ToArray();
+
+            if (errors.Length > 0)
+            {
+                return errors;
+            }
+
+            return Result.Success();
+        }
+
+        private void RegisterEvents()
+        {
+            _foregroundProcessChanged.Event += (sender, @event) =>
+            {
+                _profileByApplication.TryGetValue(@event.ProcessName.ToLower(), out var profile);
+                if (profile == null)
+                    return;
+                SwitchAudio(profile, @event.ProcessId);
+            };
+
+            WindowsAPIAdapter.HotKeyPressed += (sender, args) =>
+            {
+                _profileByHotkey.TryGetValue(args.HotKeys, out var profile);
+                if (profile == null)
+                    return;
+                SwitchAudio(profile);
+            };
+        }
+
+        private void SwitchAudio(ProfileSetting profile, uint processId = 0)
+        {
+            foreach (var device in new[] {profile.Playback, profile.Recording}.Where((info) => info != null))
+            {
+                if (processId != 0)
+                {
+                    _audioSwitcher.SwitchProcessTo(
+                        device.Id,
+                        ERole.ERole_enum_count,
+                        (EDataFlow) device.Type,
+                        processId);
+                }
+
+                //Always switch the default device.
+                //Easy way to be sure a notification will be send.
+                //And to be consistent with the default audio device.
+                _audioSwitcher.SwitchTo(device.Id, ERole.ERole_enum_count);
+            }
+        }
+
+        /// <summary>
+        /// Add a profile to the system
+        /// </summary>
+        /// <param name="profile"></param>
+        /// <returns></returns>
+        public Result<string, VoidSuccess> AddProfile(ProfileSetting profile)
+        {
+            if (string.IsNullOrEmpty(profile.ProfileName))
```

Comment:
Move the error handling to it's own method, if everything is fine continue inside Result().